media/src/main/java/io/wcm/handler/media/impl/MediaBuilderImpl.java

```diff
@@ -92,10 +100,19 @@ private static MediaComponentPropertyResolver getMediaComponentPropertyResolver(
     }
   }
 
-  MediaBuilderImpl(String mediaRef, MediaHandlerImpl mediaHandler) {
+  MediaBuilderImpl(String mediaRef, Resource contextResource, MediaHandlerImpl mediaHandler,
+      @Nullable ComponentPropertyResolverFactory componentPropertyResolverFactory) {
     this.resource = null;
```

Comment:
this should be set to the context resources - it's passed down to media handler and may be required later on in the asset processing, e.g. in the markup builders.

Reply:
When I do that, this situation will raise the exception
```
// validate parameters
    if (this.resource != null && this.mediaRef != null) {
      throw new IllegalArgumentException("Set resource or media ref, not both.");
    }
```
[in MediaRequest](https://github.com/wcm-io/wcm-io-handler/blob/develop/media/src/main/java/io/wcm/handler/media/MediaRequest.java#L112)

Removing that test does not break any unit tests in the wcm-io-handler bunde, but I still hesitate to make that change. Do you think it is safe?

Reply:
yes, i think that's fine

Reply:
ok done: Removed the Test and set the resource field to the contextResource